### PR TITLE
refactor(robot-server): Test command routes as regular Python functions

### PR DESCRIPTION
**Pull request by SyntaxColoring**
# Overview

Simplifies `robot-server/tests/sessions/router/test_commands_router.py`, per the philosophy laid out in #8394.

# Review requests

Make sure the new tests look functionally equivalent to the old ones, except that we're no longer testing that we correctly serialize and deserialize.

# Risk analysis

Low. Changes are only to tests.

### CODE REVIEW

**Review by mcous — APPROVED** [review]
Nice, LGTM

File: robot-server/tests/sessions/router/test_commands_router.py [hunk 1]
```diff
@@ -17,76 +11,48 @@
...
-            session_store=matchers.Anything(),
-            engine_store=matchers.Anything(),
-        ),
-    ).then_return(ResponseModel(data=session_response))
-
-    response = await async_client.get("/sessions/session-id/commands")
+    response = await get_session_commands(session=session_response)
 
-    verify_response(response, expected_status=200, expected_data=[command_summary])
+    assert response.data == [command_summary]
```
Comment (SyntaxColoring) [inline robot-server/tests/sessions/router/test_commands_router.py:50]: Note that we no longer explicitly test that the response code is 200. This matches what I think we did in #8258 (see [this file](https://github.com/Opentrons/opentrons/pull/8258/files#diff-b83d98aae76edb1534ad0db3204a30cb6c4830f6e4d1865c414aafa511c2c5d1)), but I'm not sure if it's right. Not all successes are `200`: `200` might be distinct from `201`, for example, and that might be important to test.

File: robot-server/tests/sessions/router/test_commands_router.py [hunk 2]
```diff
@@ -117,10 +84,7 @@ def test_get_session_command_missing_command(
...
-
-    verify_response(
-        response,
-        expected_status=404,
-        expected_errors=CommandNotFound(detail=str(key_error)),
-    )
+    with pytest.raises(ApiError) as exc_info:
+        await get_session_command(commandId="command-id", engine_store=engine_store)
+    assert exc_info.value.status_code == 404
+    assert exc_info.value.content["errors"][0]["detail"] == "oh no"
```
Comment (SyntaxColoring) [inline robot-server/tests/sessions/router/test_commands_router.py:90]: This feels weird to me, but I don't know how else we'd do it.

Comment (mcous) [reply robot-server/tests/sessions/router/test_commands_router.py:90]: I agree, I think it's a sign we need to continue working on our custom error handling. Ideally, I think, we shift things around so the `ApiError` contains the Pydantic model (with our error ID, etc.) as structured data - or something along those lines

File: robot-server/tests/sessions/router/test_commands_router.py (hunk 1 above)
Comment (mcous) [reply robot-server/tests/sessions/router/test_commands_router.py:50]: Yeah, I had the same thought in the protocols test rewrite, but I think it's the best we can do given how FastAPI insists on setting the response status code - in the route decorator.

Given that, I think the more valuable test to write for checking the success status code is a tavern test. Let's write a ticket for that and sync up with @Corey-ONeal

**Pull request merged**